DynamicTablesPkg/Include/ArchCommonNameSpaceObjects.h

```diff
@@ -50,6 +51,8 @@ typedef enum ArchCommonObjectID {
   EArchCommonObjPccSubspaceType5Info,           ///< 24 - Pcc Subspace Type 5 Info
   EArchCommonObjPsdInfo,                        ///< 25 - P-State Dependency (PSD) Info
   EArchCommonObjTpm2InterfaceInfo,              ///< 26 - TPM Interface Info
+  EArchCommonObjSpmiInterfaceInfo,              ///< 27 - SPMI Interface Info

```

Comment:
Unless one of the associated objects can be used separately, I think they can be merged into one objects

Reply:
I separated them intentionally to reduce the platform configuration effort.
1) EArchCommonObjSpmiInterfaceInfo is used by all modern platforms and BIOSes.
2) EArchCommonObjSpmiInterruptDeviceInfo is legacy and currently not used by any platform.
    All the platform sets these values as zero(0).

Reply:
I think it would be good to add the above information as a comment in the generator. 
For example, it would be good to add a comment similar to https://github.com/tianocore/edk2/blob/master/DynamicTablesPkg/Library/Acpi/Arm/AcpiGtdtLibArm/GtdtGenerator.c#L24-L33 
with the explanation you have above.